umap/management/commands/import_pictograms.py

```diff
@@ -7,40 +7,61 @@
-                            help='Update picto if it already exists.')
+        parser.add_argument("path")
+        parser.add_argument(
+            "--attribution",
+            required=True,
+            help="Attribution of the imported pictograms",
+        )
+        parser.add_argument(
+            "--extensions",
+            help="Optional list of extensins to process",
+            nargs="+",
+            default=[".svg"],
+        )
+        parser.add_argument(
+            "--exclude",
+            help="Optional list of files or dirs to exclude",
+            nargs="+",
+            default=["font"],
+        )
+        parser.add_argument(
+            "--force", action="store_true", help="Update picto if it already exists."
+        )
 
     def handle(self, *args, **options):
-        path = options['path']
-        attribution = options['attribution']
-        suffix = options['suffix']
-        force = options['force']
-        for filename in os.listdir(path):
-            if filename.endswith("-24.png"):
-                name = self.extract_name(filename)
-                if suffix:
-                    name = '{name}{suffix}'.format(name=name, suffix=suffix)
+        self.path = Path(options["path"])
+        self.attribution = options["attribution"]
+        self.extensions = options["extensions"]
+        self.force = options["force"]
+        self.exclude = options["exclude"]
+        self.handle_directory(self.path)
+
+    def handle_directory(self, path):
+        for filename in path.iterdir():
+            if filename.name in self.exclude:
+                continue
+            if filename.is_dir():
+                self.handle_directory(filename)
+                continue
+            if filename.suffix in self.extensions:
+                name = filename.stem
                 picto = Pictogram.objects.filter(name=name).last()
                 if picto:
-                    if not force:
-                        self.stdout.write(u"⚠ Pictogram with name '{name}' already exists. Skipping.".format(name=name))  # noqa
+                    if not self.force:
+                        self.stdout.write(
+                            f"⚠ Pictogram with name '{name}' already exists. Skipping."
+                        )
                         continue
                 else:
                     picto = Pictogram()
                     picto.name = name
-                filepath = os.path.join(path, filename)
-                with open(filepath, 'rb') as f:
-                    picto.attribution = attribution
-                    picto.pictogram.save(filename, File(f), save=True)
-                    self.stdout.write(u"✔ Imported pictogram {filename}.".format(filename=filename))  # noqa
-
-    def extract_name(self, filename):
-        return filename[:-7].replace('-', ' ')
+                if (path.name != self.path.name):  # Subfolders only
+                    picto.category = path.name
```

Comment:
```suggestion
                    picto.category = path.name.title()
```

Looks tempting but it doesn't work well with ALLCAPS categories (acronyms for instance).

Reply:
```suggestion
                    if path.name.upper() != path.name:
                        picto.category = path.name.title()
```
A bit ugly and I probably miss some cases…

Reply:
Yep, I'd better let people name the folder as they want, no ?